### ISSUE TITLE
Add user-friendly wrapper functions

### ISSUE THREAD

**Issue by florisvdh** [rsbivand/rgrass#89]
The addition of user-friendly wrappers was discussed last GRASS GIS Community Meeting (2023) - see discussion https://github.com/rsbivand/rgrass/discussions/78:

> TO DO: make an issue with proposed new function names, usage (new arguments with defaults), and an indication of 'becoming internal'. So also including old function names that are kept (internally).

The referred discussion contains more details. We're planning to contribute to some of the discussed **rgrass** topics at the upcoming [Meeting](https://grasswiki.osgeo.org/wiki/GRASS_Community_Meeting_Prague_2024), which starts this week.

Here follow some very rough ideas to give this a start and get some first feedback already. This definitely needs more thought and experimentation while implementing, e.g. through vignettes and examples. That also includes the arguments of below functions. I don't want to push anything, it's all open to change.

Note that the _underlying_ functions of the wrappers would be the current unchanged functions.

- They wouldn't be internalized once all new exported functions are in place, but only after a much longer period.
- Functions not covered below would not be covered by exported wrapper functions or aliases and eventually be internalized, such as the functions to set & get package specific options, but this also needs more thought (now assuming they're for internal use only, but this needs more investigation at least for me).

Proposition: eventually use `grass_` as prefix for all exported functions. (in **qgisprocess** package the `qgis_` prefix is used)

### Main functions (names are the wrappers)

`grass_init()`
`grass_vect()` (to read vector layer; name inspired by **terra**)
`grass_rast()` (to read raster; name inspired by **terra**)
`grass_write()`
`grass_exec()`

(`grass_write()` could detect whether it needs `write_VECT()` or `write_RAST()` as backend based on the inputted R object, and check whether the appropriate (and no inappropriate) arguments are present. If such integration proves too difficult or user-unfriendly, we can still split into e.g. `grass_write_vect` and `grass_write_rast()` but a short name might be appealing)

### Export; consider `grass_` function names for wrappers/aliases

`gmeta()`
`getLocationProj()`
`gmeta2grd()`
`vect2neigh()`
`vInfo()`
`vColumns()`
`vDataCount()`

### Advanced helpers (consider exporting too, in that case: consider `grass_` function names for wrappers/aliases)

`get.GIS_LOCK()`
`set.GIS_LOCK(pid)`
`unset.GIS_LOCK()`
`unlink_.gislock()`
`remove_GISRC()`

@rsbivand obviously you're best placed to assess the sensibility of this first attempt, and improve upon it.

**Comment by rsbivand** [comment]
Please take the lead. Personally, I do not grasp why function names matter, as they are only tokens, and any user should be able to learn the tokens needed for their workflows. However, since you are closer to what users expect, that's fine.

When may someone replace me as maintainer? I would really have appreciated help with #88 which remains open - could anyone with Ubuntu 22 LTS check whether the 8.3.2 binary isn't broken?

**Comment by ecodiv** [comment]
> I would really have appreciated help with #88 which remains open - could anyone with Ubuntu 22 LTS check whether the 8.3.2 binary isn't broken?

It doesn't seem so, I have no problem, and also [florisvdh](https://github.com/florisvdh) reports to have no issues.

**Comment by stevenpawley** [comment]
Thanks @adamlilith, I'm looking at `fasterRaster` to make sure there isn't any duplication, however, adding friendlier function synonyms is something that is on my roadmap (which I'll be proposing soon), but I think the purpose/workflow will be quite different.

Just checking my understanding, but it looks like `fasterRaster` is designed for working with data that primarily lives outside of GRASS, i.e., reading via `terra::rast` / `terra::vect` (or creating a GRaster/GVector directly from a file path) and then cycling it through GRASS for geoprocessing?

I'm mostly interested in making it smoother to work with GRASS itself via R, not mimicking `terra` functions, but taking on more of the concepts from `pygrass`. Currently, `rgrass` is similar to using the older  `grass.script.run_command()` approach, which requires the user to memorize all of the GRASS tool details; I'd like to have something much closer to the more modern approaches in `pygrass` or the new `grass.tools`.

**Comment by adamlilith** [comment]
@stevenpawley Yes, I think you have the gist of `fasterRaster`. My goal was to lower the curve for people already familiar with GIS in R using `terra` (and a bit `sf`). It sounds like you easier-to-implement wrappers for some existing but also some new functionality. Best of luck!
